graphdatascience/query_runner/protocol/project_protocols.py

```diff
@@ -31,12 +33,16 @@ def run_projection(
 
     @staticmethod
     def select(protocol_version: ProtocolVersion) -> "ProjectProtocol":
-        return {ProtocolVersion.V1: ProjectProtocolV1(), ProtocolVersion.V2: ProjectProtocolV2()}[protocol_version]
+        return {
+            ProtocolVersion.V1: ProjectProtocolV1(),
+            ProtocolVersion.V2: ProjectProtocolV2(),
+            ProtocolVersion.V3: ProjectProtocolV3(),
+        }[protocol_version]
```

Comment:
not changed in this PR, but we could improve the error thrown to the user.
Could happen if the client does not know the servers version (client too old).

```suggestion
        return {
            ProtocolVersion.V1: ProjectProtocolV1(),
            ProtocolVersion.V2: ProjectProtocolV2(),
            ProtocolVersion.V3: ProjectProtocolV3(),
        }.get(protocol_version)
```
this would return an Optional[ProjectProtocol]

Reply:
Makes sense, but I think we should tackle that in a PR that is focused on the UX for outdated client versions, as you suggested